src/dotenv/main.py

```diff
@@ -90,9 +90,12 @@ def set_as_environment_variables(self, override=False):
         Load the current dotenv as system environemt variable.
         """
         for k, v in self.dict().items():
-            if k in os.environ and not override:
+            if (k is None or v is None) or (k in os.environ and not override):
                 continue
-            os.environ[to_env(k)] = to_env(v)
+            key = to_env(k)
+            value = to_env(v)
+            if key and value:
+                os.environ[key] = value
```

Comment:
This is quite complicated.  What about simplifying this to the following?

```python
            if k in os.environ and not override:
                continue
            if v is not None:
                os.environ[to_env(k)] = to_env(v)
```

Reply:
Great!
That was me juggling with `mypy` errors and trying to test everything with `None` :smile: